avro/src/main/java/com/fasterxml/jackson/dataformat/avro/schema/ArrayVisitor.java

```diff
@@ -35,8 +37,8 @@ public Schema builtAvroSchema() {
             throw new IllegalStateException("No element schema created for: "+_type);
         }
         Schema schema = Schema.createArray(_elementSchema);
-        if (_type.isArrayType()) {
-            schema.addProp(AVRO_SCHEMA_PROP_CLASS, _type.toCanonical());
+        if (!_type.hasRawClass(List.class)) {
+            schema.addProp(AVRO_SCHEMA_PROP_CLASS, AvroSchemaHelper.getTypeId(_type));
```

Comment:
The actual behavior required to be compatible with the apache implementation is "If a `GenericData.Array` cannot be assigned to the type, then it must include a type ID" or it won't deserialize properly. This check resulted in slightly cleaner schemas since `List` is a reasonably common field type and doesn't require type information, but we could just do what the Apache implementation does and always include the type information, as it shouldn't hurt anything (the tests will verify this).